### PR TITLE
Issue #6778: Move css files to hook_css_alter()

### LINKED ISSUE #6778: Custom menu breakpoint makes parent Basis CSS load after the child theme skin CSS
## Description of the bug

When I set a custom menu breakpoint it changes the look of the menu. I believe it's because of this in basis_preprocess_page():

```
if ($config->get('menu_breakpoint') == 'custom') {
    backdrop_add_css(backdrop_get_path('theme', 'basis') . '/css/component/menu-dropdown.breakpoint.css', array(
      'group' => CSS_THEME,
      'every_page' => TRUE,
    ));
    backdrop_add_css(backdrop_get_path('theme', 'basis') . '/css/component/menu-dropdown.breakpoint-queries.css', array(
      'group' => CSS_THEME,
      'every_page' => TRUE,
      'media' => 'all and (min-width: ' . $config->get('menu_breakpoint_custom') . ')',
    ));
  }
```

I believe it's placing those two files so they get referenced after the child theme. So I child theme would have to *also* load it's own versions of these files in a hook_preprocess_page().

Original issue for this functionality: https://github.com/backdrop/backdrop-issues/issues/4663

### PR DESCRIPTION
**Pull request by herbdool**
Fixes https://github.com/backdrop/backdrop-issues/issues/6778

### CODE REVIEW

File: core/themes/basis/template.php [hunk 1]
```diff
@@ -8,11 +8,24 @@
...
 function basis_css_alter(&$css) {
-  // Remove Basis' `/css/component/menu-dropdown.css` if using a custom
-  // breakpoint.
-  if (config_get('menu.settings', 'menu_breakpoint') == 'custom') {
+  // Remove Basis' `/css/component/menu-dropdown.css` and add breakpoint files
+  // if using a custom breakpoint.
+  $config = config('menu.settings');
+  if ($config->get('menu_breakpoint') == 'custom') {
     $path = backdrop_get_path('theme', 'basis');
+    $weight = isset($css[$path . '/css/component/menu-dropdown.css']['weight']) ? $css[$path . '/css/component/menu-dropdown.css']['weight'] : 0;
```
Comment (quicksketch) [inline core/themes/basis/template.php]: Should this whole conditional be hinged of if this CSS file is added to the page?
```suggestion
  if (isset($css[$path . '/css/component/menu-dropdown.css']) && $config->get('menu_breakpoint') == 'custom') {
    $dropdown_css = $css[$path . '/css/component/menu-dropdown.css'];
    $path = backdrop_get_path('theme', 'basis');
```

Then reference `$dropdown_css` in the calculations below? That would look cleaner than using `$css[$path . '/css/base.css']` as the template.

File: core/themes/basis/template.php [hunk 2]
```diff
@@ -8,11 +8,24 @@
...
     unset($css[$path . '/css/component/menu-dropdown.css']);
+
+    $weight += 0.0001;
+    $css[$path . '/css/component/menu-dropdown.breakpoint.css'] = $css[$path . '/css/base.css'];
```
Comment (herbdool) [inline core/themes/basis/template.php]:
```suggestion
$css[$path . '/css/component/menu-dropdown.breakpoint.css'] = $dropdown_css;
```

File: core/themes/basis/template.php [hunk 3]
```diff
@@ -8,11 +8,24 @@
...
+    $css[$path . '/css/component/menu-dropdown.breakpoint.css']['weight'] = $weight;
+    $css[$path . '/css/component/menu-dropdown.breakpoint.css']['data'] = $path . '/css/component/menu-dropdown.breakpoint.css';
+
+    $weight += 0.0001;
+    $css[$path . '/css/component/menu-dropdown.breakpoint-queries.css'] = $css[$path . '/css/base.css'];
```
Comment (herbdool) [inline core/themes/basis/template.php]:
```suggestion
$css[$path . '/css/component/menu-dropdown.breakpoint-queries.css'] = $dropdown_css;
```

File: core/themes/basis/template.php (hunk 1 above)
Comment (herbdool) [reply core/themes/basis/template.php]: Good idea. I was trying to think of what file would likely remain, but this is better. Don't do anything if menu-dropdown.css isn't there.

File: core/themes/basis/template.php [hunk 4]
```diff
@@ -8,11 +8,25 @@
  * Implements hook_css_alter().
  */
 function basis_css_alter(&$css) {
-  // Remove Basis' `/css/component/menu-dropdown.css` if using a custom
-  // breakpoint.
-  if (config_get('menu.settings', 'menu_breakpoint') == 'custom') {
-    $path = backdrop_get_path('theme', 'basis');
+  // Remove Basis' `/css/component/menu-dropdown.css` and add breakpoint files
```
Comment (quicksketch) [inline core/themes/basis/template.php]:
```suggestion
  // Remove the Basis css/component/menu-dropdown.css and add breakpoint files
```

**Comment by quicksketch** [comment]
Merged into 1.x and 1.29.x. Thanks @herbdool!

**Pull request merged**